### PR TITLE
task done

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

**Review by fyarkiy — APPROVED** [review]
Could you check if possible to re-factor Services

File: src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -46,7 +46,12 @@ public Order update(Order order) {
...
 
     @Override
-    public boolean delete(Long id) {
+    public boolean deleteById(Long id) {
         return Storage.orders.removeIf(o -> o.getId().equals(id));
     }
+
+    @Override
+    public boolean delete(Order item) {
+        return Storage.orders.removeIf(o -> o.equals(item));
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java]:
```suggestion
    public boolean delete(Order order) {
        return Storage.orders.remove(order);
```

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 2]
```diff
@@ -39,7 +39,12 @@ public Product update(Product product) {
...
 
     @Override
-    public boolean delete(Long id) {
+    public boolean deleteById(Long id) {
         return Storage.products.removeIf(p -> p.getId().equals(id));
     }
+
+    @Override
+    public boolean delete(Product item) {
+        return Storage.products.removeIf(p -> p.equals(item));
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: same here

File: src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -45,7 +45,12 @@ public ShoppingCart update(ShoppingCart shoppingCart) {
...
 
     @Override
-    public boolean delete(Long id) {
+    public boolean deleteById(Long id) {
         return Storage.shoppingCarts.removeIf(sc -> sc.getId().equals(id));
     }
+
+    @Override
+    public boolean delete(ShoppingCart item) {
+        return Storage.shoppingCarts.removeIf(sc -> sc.equals(item));
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java]: and here

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -38,7 +38,12 @@ public User update(User user) {
...
 
     @Override
-    public boolean delete(Long id) {
+    public boolean deleteById(Long id) {
         return Storage.users.removeIf(u -> u.getId().equals(id));
     }
+
+    @Override
+    public boolean delete(User item) {
+        return Storage.users.removeIf(u -> u.equals(item));
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]: same

File: src/main/java/com/internet/shop/dao/interfaces/GenericDao.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.Optional;
+
+public interface GenericDao<T, K> {
+    T create(T item);
+
+    Optional<T> get(K id);
+
+    List<T> getAll();
+
+    T update(T product);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/interfaces/GenericDao.java]:
```suggestion
    T update(T item);
```

**Review by RomanDubovskyi — COMMENTED** [review]
btw, it's better to have travis connected, so it will take care of codestyle checks

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**